models/cryptopunks/ethereum/cryptopunks_ethereum_current_listings.sql

```diff
@@ -8,74 +8,65 @@
-                    else 'Private Listing'
-                end as event_sub_type
-            , `minValue`/1e18 as listed_price
-            , `toAddress` as listing_offered_to 
+with all_listing_events as (
+    select  punk_id
+            , event_type
+            , case  when event_type = 'Offered' and to is null then 'Public Listing'
+                    when event_type = 'Offered' and to is not null then 'Private Listing'
+                else 'Listing Withdrawn' end as event_sub_type
+            , eth_amount as listed_price
+            , to as listing_offered_to
             , evt_block_number
             , evt_index
             , evt_block_time
             , evt_tx_hash
-    from {{ source('cryptopunks_ethereum','CryptoPunksMarket_evt_PunkOffered') }}
-)
-, all_no_longer_for_sale_events (
-    select  `punkIndex` as punk_id
-            , 'No Longer For Sale' as event_type
-            , case when evt_tx_hash in (select evt_tx_hash from cryptopunks_ethereum.CryptoPunksMarket_evt_PunkBought) then 'Punk Bought'
-                    else 'Other'
-                end as event_sub_type
-            , null as listed_price
-            , null as listing_offered_to
-            , evt_block_number
-            , evt_index
-            , evt_block_time
-            , evt_tx_hash
-    from {{ source('cryptopunks_ethereum','CryptoPunksMarket_evt_PunkNoLongerForSale') }}
+    from {{ ref('cryptopunks_ethereum_punk_offer_events') }}
 )
 , all_buys as (
-    select  `punkIndex` as punk_id
+    select  token_id as punk_id
             , 'Punk Bought' as event_type
             , 'Punk Bought' as event_sub_type
-            , null as listed_price
-            , null as listing_offered_to
-            , evt_block_number
+            , cast(NULL as double) as listed_price
+            , cast(NULL as varchar(5)) as listing_offered_to
+            , block_number as evt_block_number
             , evt_index
-            , evt_block_time
-            , evt_tx_hash
-    from {{ source('cryptopunks_ethereum','CryptoPunksMarket_evt_PunkBought') }}
+            , block_time as evt_block_time
+            , tx_hash as evt_tx_hash
+    from {{ ref('cryptopunks_ethereum_trades') }}
 )
 , all_transfers as (
-    select  `punkIndex` as punk_id
+    select  punk_id
             , 'Punk Transfer' as event_type
             , 'Punk Transfer' as event_sub_type
-            , null as listed_price
-            , null as listing_offered_to
+            , cast(NULL as double) as listed_price
+            , cast(NULL as varchar(5)) as listing_offered_to
             , evt_block_number
             , evt_index
             , evt_block_time
             , evt_tx_hash
-    from {{ source('cryptopunks_ethereum','CryptoPunksMarket_evt_PunkTransfer') }}
+    from {{ ref('cryptopunks_ethereum_punk_transfers') }}
+)
+, latest_eth_price as (
+    select price
+    from prices.usd 
```

Comment:
can you add the jinja syntax here?

Reply:
sorry i wasn't clear -- not an incremental filter, this is a view, but to make `prices.usd` a source jinja syntax. you have it hardcoded :) it helps with dependencies in order of operations

Reply:
Ah sorry I missed that - yes will fix now!